tests/post/steps/test_volume.py

```diff
@@ -31,13 +32,32 @@ def test_deploy_operator(host):
 def test_volume_creation(host):
     pass
 
+@scenario('../features/volume.feature', 'Test volume deletion (sparseLoopDevice)')
+def test_volume_deletion(host):
+    pass
+
+# }}}
+
+# Given {{{
+
+@given(parsers.parse("a Volume '{name}' exists"))
+def volume_exist(host, name, k8s_custom_client):
+    volume = _volume_exist(host, name, k8s_custom_client)
+    if volume is None:
+        _create_volume(host, name, k8s_custom_client)
+
 # }}}
 # When {{{
 
 @when(parsers.parse("I create a Volume '{name}'"))
 def create_volume(host, name, k8s_custom_client):
     _create_volume(host, name, k8s_custom_client)
 
+
+@when(parsers.parse("I delete the Volume '{name}'"))
+def delete_volume(host, name, k8s_custom_client):
+    _delete_volume(host, name, k8s_custom_client)
```

Comment:
`_delete_volume` is only used here, so I don't think you need an helper.

Simply write:

```python
@when(parsers.parse("I delete the Volume '{name}'"))
def delete_volume(host, name, k8s_custom_client):
        k8s_custom_client.delete_cluster_custom_object(
            group="storage.metalk8s.scality.com",
            version="v1alpha1",
            plural="volumes",
            name=name,
            body=kubernetes.client.V1DeleteOptions(),
            grace_period_seconds=0
        )
```

We don't need to handle exception here, if something fail the test should fail.

Reply:
yes we do not need the helper yet , but the next test case makes use of a delete function definition hence the helper

Reply:
> the next test case makes use of a delete function definition hence the helper

Then introduce it in the next test case.
I'm not even sure the next test case use it actually.

The next test case use ` When I delete the Volume 'volume3'` which is exactly what we use here. No code to add for that.